### PR TITLE
Telepads are now upgradeable among other things

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
Telepads are now machineries with components
Components determine the amount of teleports before recalibration is required
Blacklisted telepad and telepad control consoles from being replicated with mechanic analyzers because I have to look at the code in the future to make linking not as bad
If you have T4 components you won't need to recalibrate at all
Telepad console now has a cool text that shows you if it needs recalibration or not

:cl:
 * tweak: Telepads can now be upgraded. The amount of teleports before recalibration is required depends on the quality of the components.
 * rscadd: The telepad console now tells you if it needs to recalibrate or not.

### CODE REVIEW

**Comment by BomberBro** [comment]
Should require T4 to never require recalibration, that's immensely powerful

**Comment by GreatBigFailure** [comment]
Post new UI. Also good change but Bomber is right about T4

**Comment by boy2mantwicethefam** [comment]
@BomberBro kay

@GreatBigFailure 
image

**Comment by SonixApache** [comment]
conflicts

File: code/modules/telesci/telepad.dm [hunk 1]
```diff
@@ -8,14 +8,40 @@
 	use_power = 1
 	idle_power_usage = 200
 	active_power_usage = 5000
-	machine_flags = MULTITOOL_MENU
+
+	machine_flags = MULTITOOL_MENU | SCREWTOGGLE | CROWDESTROY
```
Comment (jknpj) [inline code/modules/telesci/telepad.dm]: Needs FIXED2WORK. Yes, even if you can't WRENCHMOVE it, you can still unanchor it with enough autism.

File: code/modules/telesci/telepad.dm [hunk 2]
```diff
@@ -54,18 +80,25 @@
 		linked = null
 	..()
 
+/obj/machinery/telepad/attack_ai(mob/user)
```
Comment (jknpj) [inline code/modules/telesci/telepad.dm:83]: Woah pal, is this an undocumented balance change?

Comment (jknpj) [reply code/modules/telesci/telepad.dm:83]: Wait, this is not the console

File: code/modules/telesci/telepad.dm [hunk 3]
```diff
@@ -54,18 +80,25 @@
...
+	return
+
+/obj/machinery/telepad/attack_paw(mob/user)
+	return
+
+/obj/machinery/telepad/attack_hand(mob/user, ignore_brain_damage)
+	if(..())
+		return
+	if(panel_open && amplifier)
+		to_chat(user, "<span class='notice'>You carefully take \the [amplifier] from \the [src].</span>")
```
Comment (jknpj) [inline code/modules/telesci/telepad.dm:93]: >carefully
>throws into the ground

File: code/modules/telesci/telepad.dm (hunk 1 above)
Comment (jknpj) [reply code/modules/telesci/telepad.dm]:
```suggestion
	machine_flags = MULTITOOL_MENU | SCREWTOGGLE | CROWDESTROY | FIXED2WORK
```

**Pull request merged**